### ISSUE TITLE
[vrv] Subtitle Extraction

### ISSUE THREAD

**Issue by DaPatch** [ytdl-org/youtube-dl#19238]
### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2019.02.08*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2019.02.08**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [x] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [x] Site support request (request for adding support for a new site)
- [x] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):
- Single video: https://vrv.co/watch/G69V9KEDY/Regarding-My-Unforseen-Family-Crisis
- Single video: https://vrv.co/watch/GRMEEK74Y/Rocket-Power:New-Squid-on-the-Block-Down-The-Drain

Note that **youtube-dl does not support sites dedicated to [copyright infringement](https://github.com/rg3/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free)**. In order for site support request to be accepted all provided example URLs should not violate any copyrights.

---

### Description of your *issue*, suggested solution and other information

VRV extractor does not always get subs. Bothers me to the extent that I found the path to resolving.
I want to preface that Python is not something I am familiar with.

However I think adding the following to line 182 of vrv.py will increase subtitle extraction rate:
```python
for subtitle in streams_json.get('captions', {}).values():
	subtitle_url = subtitle.get('url')
	if not subtitle_url:
		continue
	subtitles.setdefault(subtitle.get('locale', 'en-US'), []).append({
		'url': subtitle_url,
		'ext': subtitle.get('format', 'vtt'),
	})
```
Basically your checking for a 'subtitles' key when you should be looking for a 'captions' key for these videos. Not sure if changed, different for different content providers, or what. Think checking for both key's values will suffice for extracting more subs across the site.

**Comment by darkhelmet2016** [comment]
But it still needs to be able to find subtitles as well since since shows like anime use ass but TV shows would use vtt, so wouldn't it need an argument to find either or?

**Comment by DaPatch** [comment]
Could always "attempt" to extract and save all the ones it extracts, as I believe the code I provide does. Shouldn't require a command line argument as long as you are telling youtube-dl to save subs.

**Comment by darkhelmet2016** [comment]
That line in py you provided is for one kind of sub vtt the argument line for either or would your line only grabs closed caption subs as each provider uses different subtitles, some vtt, some ass and some srt. You are asking it to "get" cc vtt subs.

**Comment by LostOnTheLine** [comment]
I know some places might use them interchangeably, but Closed Captions are different from Subtitles. Closed Captioning actually means it's a text of what you should be hearing, so if they are speaking Japanese English words would not be Closed Captions. I know for funimation if you have Dubbed video it does not allow subtitles, but instead has CC. Annoying for people like me who'd like the dubbed audio with translated text as opposed to just text of what you're hearing. But as CC is intended for those who can't hear it also included sound effects & often names of who is talking, especially when they are off screen or you visually wouldn't be able to clearly tell. Many movies will have a CC as well as a Subtitle track in the original language, & that's the difference.
